contracts/gateway/GatewayBase.sol

```diff
@@ -0,0 +1,395 @@
+pragma solidity ^0.4.23;
+
+import './GatewayLib.sol';
+import "./CoreInterface.sol";
+import "./SafeMath.sol";
+import "./MessageBus.sol";
+
+/**
+ *  @title GatewayBase contract.
+ *
+ *  @notice GatewayBase contains general purpose functions shared between
+ *  gateway and co-gateway contract.
+ */
+contract GatewayBase {
+
+    using SafeMath for uint256;
+    /** Emitted whenever a Gateway/CoGateway contract is proven.
+     *	wasAlreadyProved parameter differentiates between first call and replay
+     *  call of proveGateway method for same block height
+     */
+    event GatewayProven(
+        address _gateway,
+        uint256 _blockHeight,
+        bytes32 _storageRoot,
+        bool _wasAlreadyProved
+    );
+
+
+    /* Struct */
+
+    /**
+     * ActiveProcess stores the information related to in progress process
+     * like stake/mint unstake/redeem.
+     */
+    struct ActiveProcess {
+
+        /** latest message hash. */
+        bytes32 messageHash;
+
+        /** Outbox or Inbox process. */
+        MessageBus.MessageBoxType messageBoxType;
+    }
+
+
+    bytes32 constant STAKE_TYPEHASH = keccak256(
+        abi.encode(
+            "Stake(uint256 amount,address beneficiary,MessageBus.Message message)"
+        )
+    );
+
+    bytes32 constant REDEEM_TYPEHASH = keccak256(
+        abi.encode(
+            "Redeem(uint256 amount,address beneficiary,MessageBus.Message message)"
+        )
+    );
+    bytes32 constant GATEWAY_LINK_TYPEHASH = keccak256(
+        abi.encode(
+            "GatewayLink(bytes32 messageHash,MessageBus.Message message)"
+        )
+    );
+
+    /* constants */
+    /** Position of message bus in the storage */
+    uint8 constant MESSAGE_BOX_OFFSET = 1;
+    /** Penalty in bounty amount percentage charged to staker on revert staking
```

Comment:
New line after L64.

If the comment doesn't fit in one line then split it in to multiline (i.e comment will start from L66)

Reply:
Done